src/clr-angular/forms/datalist/providers/datalist-controller.service.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { Injectable } from '@angular/core';
+
+@Injectable()
+export class DatalistControllerService {
+  private _dataItems: string[] = [];
```

Comment:
Is there any reason for having setter/getter for `dataItems` can we just go with 
```ts
public dataItems: string[] = [];
```

Reply:
for now, i guess i can keep it without getter/setter. also, i have to mention here that i use this simple service as an alternative for communication between datalist container and datalist directive. my first option was to use content child in container to receive this data input, but this approach creates a circular dependency.